co-log-core/src/Colog/Core/Class.hs

```diff
@@ -18,11 +17,12 @@ TODO: laws
 class HasLog env msg m where
     getLogAction :: env -> LogAction m msg
     setLogAction :: LogAction m msg -> env -> env
+    setLogAction = seq
+    overLogAction :: (LogAction m msg -> LogAction m msg) -> env -> env
+    overLogAction = seq
+    {-# MINIMAL getLogAction #-}
 
 instance HasLog (LogAction m msg) msg m where
     getLogAction = id
-    setLogAction = const
-
-overLogAction :: HasLog env msg m => (LogAction m msg -> LogAction m msg) -> env -> env
-overLogAction over env = setLogAction (over $ getLogAction env) env
-{-# INLINE overLogAction #-}
+    setLogAction = overLogAction . const
```

Comment:
IMO the default implementation of setLogAction should be this in the typeclass, and then we don't even have to define it in this trivial case here